### PR TITLE
Our first system test!

### PR DESCRIPTION
**Pull request by advayDev1**
This can be improved a lot, but we have to capture a beach-head first.

### CODE REVIEW

**Comment by advayDev1** [comment]
Works locally, works on Travis:
https://travis-ci.org/j2objc-contrib/j2objc-gradle/jobs/77048032#L2741

ptal @brunobowden

File: systemTests/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Ignore all the wrappers below.
+gradle/
+gradlew.bat
+# Under this directory, we do in fact need to preserve local.properties!
+!local.properties
+# Generated by prep.sh
+common/local.properties
+localJ2objcDist/
```
Comment (brunobowden) [inline systemTests/.gitignore]: What are these two ignore lines for?

Comment (advayDev1) [inline systemTests/.gitignore]: As it says, those files/directories are generated by prep.sh

File: systemTests/prep.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+set -e
+
+pushd localJ2objcDist
+
+# For developer local testing, don't keep redownloading the zip file.
+if [ ! -e j2objcDist.zip ]; then
+  curl -L https://github.com/google/j2objc/releases/download/$J2OBJC_VERSION/j2objc-$J2OBJC_VERSION.zip > j2objcDist.zip
+  unzip j2objcDist.zip
+  mv j2objc-$J2OBJC_VERSION j2objcDist
+  echo j2objc.home=$PWD/j2objcDist > ../common/local.properties
```
Comment (brunobowden) [inline systemTests/prep.sh]: Does this just append rather than replace? Probably fine for now.

Comment (advayDev1) [inline systemTests/prep.sh]: It replaces, which is correct.

File: systemTests/simple1/src/main/java/com/example/HelloWorld.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    public HelloWorld(String nameOfWorld) {
+        this.nameOfWorld = nameOfWorld;
+    }
+
+    @Override
+    public String toString() {
+        return "Hello " + nameOfWorld;
+    }
+}
```
Comment (brunobowden) [inline systemTests/simple1/src/main/java/com/example/HelloWorld.java]: Should have trailing newline

File: systemTests/prep.sh (hunk 2 above)
Comment (advayDev1) [inline systemTests/prep.sh]: append is `>>`

File: systemTests/simple1/src/main/java/com/example/HelloWorld.java (hunk 3 above)
Comment (advayDev1) [inline systemTests/simple1/src/main/java/com/example/HelloWorld.java]: done

**Comment by brunobowden** [comment]
LGTM

You should also see if you can validate a build error. E.g. by posting a PR that breaks the Hello World... both the 'java' build and then translated, e.g. using some OCNI junk.

**Comment by advayDev1** [comment]
That's a really good point.  Will do

**Comment by advayDev1** [comment]
Errors correctly fail:
https://travis-ci.org/j2objc-contrib/j2objc-gradle/jobs/77052795

merging

**Pull request merged**